src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -142078,5 +142121,658 @@ static T GetValue2<T>(Collection c, object key)
+                // (38,11): error CS0452: The type 'T' must be a reference type in order to use it as parameter 'TCO' in the generic type or method 'C.FTCO<TCO>(TCO)'
+                //         C.FTCO<T>(default).ToString(); // illegal 
+                Diagnostic(ErrorCode.ERR_RefConstraintNotSatisfied, "FTCO<T>").WithArguments("C.FTCO<TCO>(TCO)", "TCO", "T").WithLocation(38, 11),
+                // (46,9): warning CS8602: Dereference of a possibly null reference.
+                //         x0.ToString(); // 9
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x0").WithLocation(46, 9),
+                // (48,27): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //         var x1 = C.FT<TC>(default); // 10
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "default").WithLocation(48, 27),
+                // (51,28): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //         var x2 = C.FTC<TC>(default); // 11
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "default").WithLocation(51, 28),
+                // (54,28): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //         var x3 = C.FTO<TC>(default); // 12
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "default").WithLocation(54, 28),
+                // (57,29): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //         var x4 = C.FTCO<TC>(default); // 13
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "default").WithLocation(57, 29),
+                // (76,11): error CS0452: The type 'TO' must be a reference type in order to use it as parameter 'TC' in the generic type or method 'C.FTC<TC>(TC)'
+                //         C.FTC<TO> // illegal
+                Diagnostic(ErrorCode.ERR_RefConstraintNotSatisfied, "FTC<TO>").WithArguments("C.FTC<TC>(TC)", "TC", "TO").WithLocation(76, 11),
+                // (86,11): error CS0452: The type 'TO' must be a reference type in order to use it as parameter 'TCO' in the generic type or method 'C.FTCO<TCO>(TCO)'
+                //         C.FTCO<TO> // illegal
+                Diagnostic(ErrorCode.ERR_RefConstraintNotSatisfied, "FTCO<TO>").WithArguments("C.FTCO<TCO>(TCO)", "TCO", "TO").WithLocation(86, 11),
+                // (100,9): warning CS8602: Dereference of a possibly null reference.
+                //         x0.ToString(); // 14
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "x0").WithLocation(100, 9),
+                // (111,14): warning CS8625: Cannot convert null literal to non-nullable reference type.
+                //             (default); // 15
+                Diagnostic(ErrorCode.WRN_NullAsNonNullable, "default").WithLocation(111, 14)
+                );
+        }
+
+        [Fact]
+        public void DefaultParameterValue()
```

Comment:
:memo: some of the tests that I added are not so relevant to this new PR, I just moved all the new tests from my earlier PR. Seems okay to keep them.